index.js

```diff
@@ -10,7 +10,10 @@ module.exports = {
           * Always delay requires, otherwise your plugin will cause trouble
           * with db-migrates performance and generates issues to your users.
           */
-        require('typescript').register();
+        var typescriptModule = require('typescript');
+        if (typescriptModule && typescriptModule.register) {
```

Comment:
I found this informations using google, I don't have the reference right now but I can search for it.
Thinking about the code, maybe this way would be better:
```js
module.exports = Object.assign(module.exports, {

      'migrator:migration:hook:require': function() {

        try {
          /**
           * Always delay requires, otherwise your plugin will cause trouble
           * with db-migrates performance and generates issues to your users.
           */
          var typescriptModule = require('typescript');
          if (typescriptModule) {
            if (typescriptModule.register) {
              typescriptModule.register();
            }
            return {
              extensions: 'ts'
            };
          }
        } catch(e) {}
        return {};
      }
    });
```
It prevents the plugin from crashing if there is a problem with the typescript module, what do you think ?